Refactor webroot detection in resource locator

tux-rampage opened the pull request:
The current implementation breaks installations with symlinks to
directories inside the webroot (i.E. apps).

With this change both variants, directory and symlinks) will be detected
correctly.

Fixes: #6028

MorrisJobke commented:
One positive feedback already: https://github.com/nextcloud/server/issues/6028#issuecomment-326953110

MorrisJobke commented on lib/private/Template/ResourceLocator.php:
```diff
@@ -107,6 +107,51 @@ protected function appendIfExist($root, $file, $webRoot = null) {
+	 * @return string|null The web root or null on failure
+	 */
+	protected function findWebRoot($root)
+	{
```
Please put it on the end of the line before this.

tux-rampage replied:
What do you men? Putting the method to the bottom of the class?

skjnldsv replied:
He means the bracket on the end of line :)

MorrisJobke commented on lib/private/Template/ResourceLocator.php:
```diff
@@ -107,6 +107,51 @@ protected function appendIfExist($root, $file, $webRoot = null) {
+		if (!$webRoot) {
+			$realpath = realpath($root);
+
+			if ($realpath && ($realpath != $root)) {
```
Please use `!==`

MorrisJobke reviewed (approved):
Code change makes sense and still works 👍

MorrisJobke commented:
> Code change makes sense and still works 👍

Comments from above still apply of course 😉

tux-rampage commented:
Do you want me to push a squashed commit, or will it be OK to simply push the requested changes?

MorrisJobke commented:
> Do you want me to push a squashed commit, or will it be OK to simply push the requested changes?

Squashed would be awesome :)

tux-rampage committed:
Refactor webroot detection in resource locator

The current implementation breaks installations with symlinks to
directories inside the webroot (i.E. apps).

With this change both variants, directory and symlinks, will be detected
correctly.

Fixes: #6028
Signed-off-by: Axel Helmert <axel.helmert@luka.de>

tux-rampage commented:
Added style fixes as requested - squashed commit

skjnldsv reviewed (approved):
Very nice!

leonklingele-work commented on lib/private/Template/CSSResourceLocator.php:
```diff
@@ -117,38 +117,18 @@ public function append($root, $file, $webRoot = null, $throw = true, $scss = fal
 			parent::append($root, $file, $webRoot, $throw);
 		} else {
 			if (!$webRoot) {
-				$tmpRoot = realpath($root);
```
This removed var is still used in line 134 which results in error msgs: `Undefined variable: tmpRoot`

skjnldsv replied:
Nice catch, I ashamed to have missed that! 🙈

tux-rampage commented:
True, should I add another PR?

skjnldsv commented:
@tux-rampage yes :)
